File: demo/src/main/java/cc/zgeek/pagedemo/InnerVP.java
package cc.zgeek.pagedemo;

import android.support.design.widget.TabLayout;
import android.support.v4.view.ViewPager;
import android.view.ViewGroup;

import java.util.Arrays;

import cc.zgeek.pagelib.Annotation.InjectView;
import cc.zgeek.pagelib.Annotation.PageLayout;
import cc.zgeek.pagelib.IPage;
import cc.zgeek.pagelib.Page;
import cc.zgeek.pagelib.PageActivity;
import cc.zgeek.pagelib.ViewPagerPage;

/**
 * Created by flyop.
 * Change History:
 * 2017/1/29 : Create
 */

@PageLayout(R.layout.inner_vp)
public class InnerVP extends Page {

    @InjectView(R.id.layout_tab_container)
    TabLayout tabLayout;
    @InjectView(R.id.vp_container)
    ViewGroup viewGroup;

    public InnerVP(PageActivity pageActivity) {
        super(pageActivity);
    }

    @Override
    public void onViewInited() {

        ViewPagerPage innerPager = new ViewPagerPage(getContext());
        innerPager.addPages(Arrays.asList(
                new SimplePage(getContext()).setName("内0"),
                new SimplePage(getContext()).setName("内1"),
                new SimplePage(getContext()).setName("内2")));

        tabLayout.setupWithViewPager((ViewPager) innerPager.getRootView());
        viewGroup.addView(innerPager.getRootView());
        addPage(innerPager);
    }
}

File: demo/src/main/java/cc/zgeek/pagedemo/InnerVP.kt
package cc.zgeek.pagedemo

import android.os.Bundle
import android.support.design.widget.TabLayout
import android.support.v4.view.ViewPager
import android.view.ViewGroup

import java.util.Arrays

import cc.zgeek.pagelib.Annotation.InjectView
import cc.zgeek.pagelib.Annotation.PageLayout
import cc.zgeek.pagelib.Page
import cc.zgeek.pagelib.PageActivity
import cc.zgeek.pagelib.ViewPagerPage

/**
 * Created by flyop.
 * Change History:
 * 2017/1/29 : Create
 */

@PageLayout(R.layout.inner_vp)
class InnerVP(pageActivity: PageActivity) : Page(pageActivity) {

    @InjectView(R.id.layout_tab_container)
    internal lateinit var tabLayout: TabLayout
    @InjectView(R.id.vp_container)
    internal lateinit var viewGroup: ViewGroup

    override fun onViewInitialized(isRestore: Boolean, args: Bundle) {
        super.onViewInitialized(isRestore, args)
        if (!isRestore) {
            val innerPager = ViewPagerPage(context)
            innerPager.addPages(Arrays.asList(
                    SimplePage.newInstance(context),
                    SimplePage.newInstance(context),
                    SimplePage.newInstance(context)))
            //            innerPager.getChildPageAt(0) = .setName("内2")
            //            innerPager.getSubChildPages()
            addPage(innerPager)
        }

        tabLayout.setupWithViewPager(getChildPageAt(0).rootView as ViewPager)
        viewGroup.addView(getChildPageAt(0).rootView)
    }

    //    @Override
    //    public Bundle onSaveInstanceState(boolean isViewInited) {
    //        Bundle bundle = super.onSaveInstanceState(isViewInited);
    //
    //    }
}
